### PR TITLE
LABS-485 add support for smarter buffer management

### PR DESCRIPTION
**Pull request by steve-1820**

### CODE REVIEW

File: src/stream/source_connectors/manual_insertion_source_connector.h [hunk 1]
```diff
@@ -22,6 +22,9 @@ class ManualInsertionSourceConnector : public SourceConnector {
   mutable Mutex _mutex =
       MONGO_MAKE_LATCH("ManualInsertionSourceConnector::_mutex");
   std::deque<DocumentSource::GetNextResult> _insertionQueue;
+
+  int _maxBufferSize = 1024 * 1024 * 150;
```
Comment (michaelcahill) [inline src/stream/source_connectors/manual_insertion_source_connector.h:26]: Is 150MB based on anything? Is there some document where we should note this, and that we might need to make it configurable in future?

**Review by michaelcahill — APPROVED** [review]
lgtm % minor question.

File: src/stream/source_connectors/manual_insertion_source_connector.h (hunk 1 above)
Comment (steve-1820) [reply src/stream/source_connectors/manual_insertion_source_connector.h:26]: @michaelcahill I've created a ticket to make this configurable: https://jira.mongodb.org/browse/LABS-496

150mb was loosely chosen as the aggregation pipeline spill to disk is 100mb.

Comment (michaelcahill) [reply src/stream/source_connectors/manual_insertion_source_connector.h:26]: 👍

**Pull request merged**